app/src/pages/inside/launchesPage/modals/moveToDebugModal/moveToDebugModal.jsx

```diff
@@ -18,33 +24,93 @@ const messages = defineMessages({
     id: 'MoveToDebugDialog.moveToDebug',
     defaultMessage: 'Are you sure you want to move launch to Debug?',
   },
+  moveToDebugMultipleText: {
+    id: 'MoveToDebugDialog.moveToDebugMultiple',
+    defaultMessage: 'Are you sure you want to move selected launches to Debug?',
+  },
+  moveToDebugFailedMessage: {
+    id: 'MoveToDebugDialog.moveToDebugFailedMessage',
+    defaultMessage: 'Failed to move to debug the launch: {message}',
+  },
+  moveToDebugSuccessMessage: {
+    id: 'MoveToDebugDialog.moveToDebugSuccessMessage',
+    defaultMessage: 'Launch has been successfully shifted to debug',
+  },
+  moveToDebugMultipleSuccessMessage: {
+    id: 'MoveToDebugDialog.moveToDebugMultipleSuccessMessage',
+    defaultMessage: 'Launches have been successfully shifted to debug',
+  },
 });
 
 @withModal('moveToDebugModal')
 @injectIntl
+@connect(
+  (state) => ({
+    url: URLS.launchUpdate(activeProjectSelector(state)),
+  }),
+  {
+    showNotification,
+  },
+)
 export class MoveToDebugModal extends Component {
   static propTypes = {
     intl: intlShape.isRequired,
+    url: PropTypes.string.isRequired,
     data: PropTypes.shape({
-      onConfirm: PropTypes.func,
+      fetchFunc: PropTypes.func,
+      ids: PropTypes.array,
     }),
+    showNotification: PropTypes.func,
   };
 
   static defaultProps = {
     data: {
-      onConfirm: () => {},
+      fetchFunc: () => {},
+      ids: [],
     },
+    showNotification: () => {},
   };
-  confirmAndClose = (closeModal) => {
-    this.props.data.onConfirm();
+
+  moveAndClose = (closeModal) => {
+    const { ids, fetchFunc } = this.props.data;
+    const entities = {};
+    ids.forEach((id) => {
+      entities[id] = {};
+      entities[id].mode = DEBUG.toUpperCase();
+    });
+    fetch(this.props.url, {
+      method: 'put',
+      data: {
+        entities,
+      },
+    })
+      .then(() => {
+        const successMessage =
+          (ids.length > 1 &&
```

Comment:
I think ternary operator will be more readable here

Reply:
Fixed